pkg/graveler/retention/expired_commits.go

```diff
@@ -113,17 +114,25 @@ func GetGarbageCollectionCommits(ctx context.Context, startingPointIterator *GCS
 				// the threshold. In either way, the PARENT is active.
 				activeMap[nextCommitID] = struct{}{}
 				delete(expiredMap, nextCommitID)
+				delete(previouslyExpiredMap, nextCommitID)
 			} else if _, ok = activeMap[nextCommitID]; !ok {
 				// If the parent commit was expired in a previous GC run (incremental GC case) we can stop because it's
 				// both expired in this run and the previous (in this path)
 				if _, ok = previouslyExpiredMap[nextCommitID]; ok {
+					// Get the previously expired commit's list of known children and add the current commit to it.
+					var nextCommitChildren = previouslyExpiredMap[nextCommitID]
+					if _, ok := nextCommitChildren[currentCommitID]; !ok {
+						nextCommitChildren[currentCommitID] = struct{}{}
+						previouslyExpiredMap[nextCommitID] = nextCommitChildren
+					}
 					break
```

Comment:
```suggestion
				if nextCommitChildren, ok := previouslyExpiredMap[nextCommitID]; ok {
					// Add current commit to previously expired commit's list of known children.
					if _, ok := nextCommitChildren[currentCommitID]; !ok {
						nextCommitChildren[currentCommitID] = struct{}{}
						previouslyExpiredMap[nextCommitID] = nextCommitChildren
					}
					break
```
But TBH not sure what "previouslyExpiredMap" really holds.

Reply:
done